qa/tasks/rabbitmq.py

```diff
@@ -0,0 +1,134 @@
+"""
+Deploy and configure RabbitMQ for Teuthology
+"""
+import contextlib
+import logging
+
+from teuthology import misc as teuthology
+from teuthology import contextutil
+from teuthology.orchestra import run
+from teuthology.exceptions import ConfigError
+from teuthology.config import config as teuth_config
+
+log = logging.getLogger(__name__)
+
+
+@contextlib.contextmanager
+def install_rabbitmq(ctx, config):
+    """
+    Downloading the RabbitMQ package.
+    """
+    assert isinstance(config, dict)
+    log.info('Installing RabbitMQ...')
+
+    for (client, _) in config.items():
+        (remote,) = ctx.cluster.only(client).remotes.keys()
+        test_dir=teuthology.get_testdir(ctx)
+
+        ctx.cluster.only(client).run(args=[
+             'sudo', 'yum', '-y', 'install', 'epel-release'
+        ])
+
+        link1 = 'https://packagecloud.io/install/repositories/rabbitmq/erlang/script.rpm.sh'
+
+        ctx.cluster.only(client).run(args=[
+             'curl', '-s', link1, run.Raw('|'), 'sudo', 'bash'
+        ])
+
+        ctx.cluster.only(client).run(args=[
+             'sudo', 'yum', '-y', 'install', 'erlang'
+        ])
+
+        link2 = 'https://packagecloud.io/install/repositories/rabbitmq/rabbitmq-server/script.rpm.sh'
+
+        ctx.cluster.only(client).run(args=[
+             'curl', '-s', link2, run.Raw('|'), 'sudo', 'bash'
+        ])
+
+        ctx.cluster.only(client).run(args=[
+             'sudo', 'yum', '-y', 'install', 'rabbitmq-server-3.8.14-1.el8'
```

Comment:
i don't think we want to hard-code the version number. we should test against the latest version

can we not get recent enough packages from the distributions? it's unfortunate that we need these extra steps, which apparently doesn't work for ubuntu?

Reply:
I actually didn't want to hard-code version number but I guess erlang only supports only rabbitmq-server version 3.8.14 at the latest. Though 3.8.16 is the latest available but it was showing some error and I asked about it on #sepia channel and they suggested this.

Reply:
did you see the error after limiting the distro to centos only?

Reply:
No, on CentOS everything is running good.

Reply:
so, since we are currently limited to centos anyway, you can just install rabbitmq-server without specifying the version